src/transport/raw/tests/TestMessageHeader.cpp

```diff
@@ -26,283 +26,282 @@
-#include <lib/support/UnitTestRegistration.h>
 #include <protocols/Protocols.h>
 #include <transport/raw/MessageHeader.h>
 
-#include <nlunit-test.h>
+#include <gtest/gtest.h>
 
 namespace {
 
 using namespace chip;
 
-void TestPacketHeaderInitialState(nlTestSuite * inSuite, void * inContext)
+TEST(TestMessageHeader, TestPacketHeaderInitialState)
 {
     PacketHeader header;
 
-    NL_TEST_ASSERT(inSuite, !header.IsSecureSessionControlMsg());
-    NL_TEST_ASSERT(inSuite, header.GetMessageCounter() == 0);
-    NL_TEST_ASSERT(inSuite, header.GetSessionId() == 0);
-    NL_TEST_ASSERT(inSuite, header.GetSessionType() == Header::SessionType::kUnicastSession);
-    NL_TEST_ASSERT(inSuite, header.IsSessionTypeValid());
-    NL_TEST_ASSERT(inSuite, !header.IsEncrypted());
-    NL_TEST_ASSERT(inSuite, !header.GetDestinationNodeId().HasValue());
-    NL_TEST_ASSERT(inSuite, !header.GetDestinationGroupId().HasValue());
-    NL_TEST_ASSERT(inSuite, !header.GetSourceNodeId().HasValue());
+    EXPECT_FALSE(header.IsSecureSessionControlMsg());
+    EXPECT_EQ(header.GetMessageCounter(), (uint32_t) 0);
+    EXPECT_EQ(header.GetSessionId(), 0);
+    EXPECT_EQ(header.GetSessionType(), Header::SessionType::kUnicastSession);
+    EXPECT_TRUE(header.IsSessionTypeValid());
+    EXPECT_FALSE(header.IsEncrypted());
+    EXPECT_FALSE(header.GetDestinationNodeId().HasValue());
+    EXPECT_FALSE(header.GetDestinationGroupId().HasValue());
+    EXPECT_FALSE(header.GetSourceNodeId().HasValue());
 }
 
-void TestPayloadHeaderInitialState(nlTestSuite * inSuite, void * inContext)
+TEST(TestMessageHeader, TestPayloadHeaderInitialState)
 {
     PayloadHeader header;
 
-    NL_TEST_ASSERT(inSuite, header.GetMessageType() == 0);
-    NL_TEST_ASSERT(inSuite, header.GetExchangeID() == 0);
-    NL_TEST_ASSERT(inSuite, header.HasProtocol(Protocols::NotSpecified));
+    EXPECT_EQ(header.GetMessageType(), 0);
+    EXPECT_EQ(header.GetExchangeID(), 0);
+    EXPECT_TRUE(header.HasProtocol(Protocols::NotSpecified));
 }
 
-void TestPacketHeaderEncodeDecode(nlTestSuite * inSuite, void * inContext)
+TEST(TestMessageHeader, TestPacketHeaderEncodeDecode)
 {
     PacketHeader header;
     uint8_t buffer[64];
     uint16_t encodeLen;
     uint16_t decodeLen;
 
     header.SetMessageCounter(123);
-    NL_TEST_ASSERT(inSuite, header.Encode(buffer, &encodeLen) == CHIP_NO_ERROR);
+    EXPECT_EQ(header.Encode(buffer, &encodeLen), CHIP_NO_ERROR);
 
     // change it to verify decoding
     header.SetMessageCounter(222).SetSourceNodeId(1).SetDestinationNodeId(2);
 
-    NL_TEST_ASSERT(inSuite, header.Decode(buffer, &decodeLen) == CHIP_NO_ERROR);
-    NL_TEST_ASSERT(inSuite, encodeLen == decodeLen);
-    NL_TEST_ASSERT(inSuite, header.GetMessageCounter() == 123);
-    NL_TEST_ASSERT(inSuite, !header.GetDestinationNodeId().HasValue());
+    EXPECT_EQ(header.Decode(buffer, &decodeLen), CHIP_NO_ERROR);
+    EXPECT_EQ(encodeLen, decodeLen);
+    EXPECT_EQ(header.GetMessageCounter(), (uint32_t) 123);
```

Comment:
Use unsigned literal instead of `C` casting (in other places too).

Reply:
Changed.